### PR TITLE
custom spinners with docs and statusline integration

### PR DESCRIPTION
**Pull request by sbromberger**
This is a clean implementation of #2793 applied to the latest master.

### CODE REVIEW

**Comment by sbromberger** [comment]
This is ready for review. cc @archseer @the-mikedavis. Thanks.

File: helix-term/src/ui/spinner.rs [hunk 1]
```diff
@@ -46,22 +64,31 @@ impl Spinner {
...
     pub fn frame(&self) -> Option<&str> {
-        let idx = (self
-            .start
-            .map(|time| Instant::now().duration_since(time))?
-            .as_millis()
-            / self.interval as u128) as usize
-            % self.count;
-
-        self.frames.get(idx).copied()
+        let idx = if self.count > 1 {
```
Comment (archseer) [inline helix-term/src/ui/spinner.rs:81]: This adds complexity. Revert the changes and make the spinner in config an Option<Spinner>. If an empty set is passed just use the default.

File: helix-view/src/editor.rs [hunk 2]
```diff
@@ -162,6 +164,10 @@ pub struct Config {
     pub rulers: Vec<u16>,
     #[serde(default)]
     pub whitespace: WhitespaceConfig,
+    /// Vector of Strings of custom spinner frames, one per element
+    pub spinner_frames: Vec<String>,
```
Comment (archseer) [inline helix-view/src/editor.rs:168]: Both of these should be nested in an object: `spinner = { frames =, interval =}`

**Review by archseer — COMMENTED** [review]
I'm still skeptical whether this is needed: it's nice to have it configurable, but will anyone actually change the default? We deliberately left this out of the original spinner PR.

File: helix-term/src/ui/spinner.rs (hunk 1 above)
Comment (archseer) [reply helix-term/src/ui/spinner.rs:81]: Alternatively: You could make get() always return &str and simply unwrap_or("")

**Comment by sbromberger** [comment]
> I'm still skeptical whether this is needed: it's nice to have it configurable, but will anyone actually change the default? 

Objectively, yes: the original PR got several positive feedback icons, and at least a few of us have been using this for months.

Why? Well, with respect, the default spinner is suboptimal under certain circumstances (and subjectively ugly): not only is the braille hard to see, but on underpowered machines or over high-latency connections, the screen refreshes take a very long time. From an accessibility perspective, having a single frame spinner (on/off) to let you know that LSP operations are in progress is much better. Since I was adding the single frame capability, it just made sense to allow full customization.

Personal anecdote: on a recent trip, over a low-bandwidth/high-latency (satellite) connection, using helix over mosh with the default spinners was a really bad experience. Switching to single-frame spinners made the experience less painful.

**Comment by sbromberger** [comment]
I'll wait to hear whether you're amenable to making this change before I address the other code comments. Thanks.

File: helix-term/src/ui/spinner.rs (hunk 1 above)
Comment (sbromberger) [reply helix-term/src/ui/spinner.rs:81]: The count is necessary since if we have a single frame we don't worry about the call to (and calculations with) `Instant::now()` since the spinner is either on with frame 0 or off. This is independent of the actual frames, so I'm not quite sure how checking an `Option` would work here.
